Use correct Istio Metadata to find out IP addresses of a proxy.

robertpanzer opened the pull request:
It looks like currently Pilot does not look at the correct metadata sent by a proxy to determine its IP address.
While the bootstrap config creates the metadata `ISTIO_META_INSTANCE_IPS` in the bootstrap config:
https://github.com/istio/istio/blob/fdefe93b83c10b016cf77aa140f77a5d0e1e90d5/pkg/bootstrap/bootstrap_config.go#L269
Pilot looks for `INSTANCE_IPS` when parsing the metadata and falls back to the second field of the node id when this elements of the metadata is unavailable:
https://github.com/istio/istio/blob/fdefe93b83c10b016cf77aa140f77a5d0e1e90d5/pilot/pkg/model/context.go#L575-L576
https://github.com/istio/istio/blob/fdefe93b83c10b016cf77aa140f77a5d0e1e90d5/pilot/pkg/model/context.go#L294-L305
This functions is called very early after receiving the request, so it looks like there is no further processing in between that strips the `ISTIO_META_` prefix.
That means it should always use this fallback right now, even though it is documented to be a fallback for older proxy versions.

So this looks like an error to me.

This PR tries to fix this by making Pilot look for the metadata `ISTIO_META_INSTANCE_IPS` instead.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/13843#" title="Author self-approved">robertpanzer</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **rshriram**

If they are not already assigned, you can assign the PR to them by writing `/assign @rshriram` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[pilot/OWNERS](https://github.com/istio/istio/blob/release-1.1/pilot/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["rshriram"]} -->

istio-testing commented:
Hi @robertpanzer. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

hzxuzhonghu commented:
You should not fix like this.  I would pass INSTANCE_IPS in node meta as other keys.

robertpanzer commented:
Wouldn't it make more sense to keep the prefix `ISTIO_META_` in the Envoy configuration?
I can understand that users should not attach labels starting with `ISTIO_META_` to their Pods.
But `INSTANCE_IPS` sounds like something that could also be configured by a user.
This would result in a conflict, because Pod labels just land literally in `node.metadata`:
```
$ kubectl get pods -n sr-test testcurl-6dd96bb67b-857xn --show-labels
NAME                        READY   STATUS    RESTARTS   AGE     LABELS
testcurl-6dd96bb67b-857xn   2/2     Running   0          3m34s   INSTANCE_IPS=abcde,app=testcurl,pod-template-hash=6dd96bb67b          
$ kubectl exec -ti -n sr-test testcurl-6dd96bb67b-857xn -c istio-proxy head /etc/istio/proxy/envoy-rev0.json                                                    

{
  "node": {
    "id": "sidecar~10.1.13.212~testcurl-6dd96bb67b-857xn.sr-test~sr-test.svc.cluster.local",
    "cluster": "testcurl.sr-test",
    "locality": {



    },
    "metadata": {"CONFIG_NAMESPACE":"sr-test","INSTANCE_IPS":"abcde","INTERCEPTION_MODE":"REDIRECT","ISTIO_META_INSTANCE_IPS":"10.1.13.212,10.1.13.212","ISTIO_PROXY_SHA":"istio-proxy:73fa9b1f29f91029cc2485a685994a0d1dbcde21","ISTIO_PROXY_VERSION":"1.1.3","ISTIO_VERSION":"1.1.5","POD_NAME":"testcurl-6dd96bb67b-857xn","app":"testcurl","istio":"sidecar","pod-template-hash":"6dd96bb67b"}
```
Apparently Pod labels are directly added to the node metadata.

howardjohn commented:
/ok-to-test

howardjohn commented on pilot/pkg/model/context.go:
```diff
@@ -561,7 +561,7 @@ const (
 
 	// NodeMetadataInstanceIPs is the set of IPs attached to this proxy
-	NodeMetadataInstanceIPs = "INSTANCE_IPS"
+	NodeMetadataInstanceIPs = "ISTIO_META_INSTANCE_IPS"
```
Do we do the stripping in bootstrap, which is where we add this to the metadata with the prefix, so this one never gets stripped?

If so, maybe a better fix is to change the bootstrap to not have the prefix, like @hzxuzhonghu said

afaik in the final config nothing should have ISTIO_META

istio-testing commented:
@robertpanzer: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-simpleTests/21566) | `/test e2e-simpleTests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

robertpanzer commented:
Yes, probably right.
The error rather seems to be that sth with `ISTIO_META_` lands in the node meta data at all.
After looking more at the code it seems like the prefix `ISTIO_META_` is used for environment variables to map them into node metadata, without prefix.
I'll update that.

I still think though that a collision is still possible, keys like `ISTIO_PROXY_SHA` can be related uniquely to Istio, but `INSTANCE_IPS`?

howardjohn commented:
Ah I misunderstood before. Is the conflict a real issue for your uses, or just a theoretical problem?

I ask not because I don't think its a legitimate problem, but the same case exists with all of the other metadata right? Its pretty uncommon to have a label like `INSTANCE_IPS` or something that would conflict. So while ideally we would probably want them all to have an istio prefix to ensure no (reasonable) conflicts, in practice it may not be worth the effort (and, more importantly, the backwards incompatibility.

If it is a problem for you though, then we can come up with a better solution

robertpanzer commented:
It's rather theoretical.
I think though that these properties should be somehow "namespaced" if different sources are just thrown together.
If this possible conflict is accepted, then I am fine with changing this accordingly so the bootstrap_config creates the property without the `ISTIO_META_` prefix.

istio-testing commented:
@robertpanzer: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19513) | `/test istio-pilot-e2e-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented:
In my opinion, at least for 1.1, we should accept the possible conflict since we don't want backwards incompatibility changes. In 1.2/beyond we could make changes, but I don't think it is worth it. I agree completely that it would be better if they were namespaced

istio-testing commented:
@robertpanzer: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19513) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-multicluster-e2e/11956) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@robertpanzer: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19513) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-multicluster-e2e/11956) | `/test istio-pilot-multicluster-e2e`
prow/release-test.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/release-test/6871) | `/test release-test`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/13843#" title="Author self-approved">robertpanzer</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **nmittler**

If they are not already assigned, you can assign the PR to them by writing `/assign @nmittler` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[pilot/OWNERS](https://github.com/istio/istio/blob/release-1.1/pilot/OWNERS)**
- **[pkg/OWNERS](https://github.com/istio/istio/blob/release-1.1/pkg/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["nmittler"]} -->

robertpanzer commented:
Updated the PR to use `INSTANCE_IPS` in the node metadata.
Additionally I added a logic to uniq the IP addresses that are written to the envoy config.
I usually see the same IP address twice in this field.
And if this field is now actually used, there's a lot of logic in Pilot that iterates over all IP addresses, which would now also run twice.

istio-testing commented:
@robertpanzer: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19513) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-multicluster-e2e/11956) | `/test istio-pilot-multicluster-e2e`
prow/release-test.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/release-test/6871) | `/test release-test`
prow/istio-unit-tests.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-unit-tests/24864) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@robertpanzer: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19513) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-multicluster-e2e/11956) | `/test istio-pilot-multicluster-e2e`
prow/release-test.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/release-test/6871) | `/test release-test`
prow/istio-unit-tests.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-unit-tests/24864) | `/test istio-unit-tests`
prow/e2e-simpleTests-cni.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-simpleTests-cni/3841) | `/test e2e-simpleTests-cni`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@robertpanzer: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19513) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-multicluster-e2e/11956) | `/test istio-pilot-multicluster-e2e`
prow/release-test.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/release-test/6871) | `/test release-test`
prow/istio-unit-tests.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-unit-tests/24864) | `/test istio-unit-tests`
prow/e2e-simpleTests-cni.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-simpleTests-cni/3841) | `/test e2e-simpleTests-cni`
prow/e2e_pilotv2_auth_sds.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio_auth_sds_e2e/5857) | `/test istio_auth_sds_e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@robertpanzer: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19513) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-multicluster-e2e/11956) | `/test istio-pilot-multicluster-e2e`
prow/release-test.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/release-test/6871) | `/test release-test`
prow/istio-unit-tests.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-unit-tests/24864) | `/test istio-unit-tests`
prow/e2e-simpleTests-cni.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-simpleTests-cni/3841) | `/test e2e-simpleTests-cni`
prow/e2e_pilotv2_auth_sds.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio_auth_sds_e2e/5857) | `/test istio_auth_sds_e2e`
prow/e2e-mixer-no_auth.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-mixer-no_auth/16502) | `/test e2e-mixer-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@robertpanzer: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-multicluster-e2e/11956) | `/test istio-pilot-multicluster-e2e`
prow/release-test.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/release-test/6871) | `/test release-test`
prow/istio-unit-tests.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-unit-tests/24864) | `/test istio-unit-tests`
prow/e2e-simpleTests-cni.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-simpleTests-cni/3841) | `/test e2e-simpleTests-cni`
prow/e2e_pilotv2_auth_sds.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio_auth_sds_e2e/5857) | `/test istio_auth_sds_e2e`
prow/e2e-mixer-no_auth.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-mixer-no_auth/16502) | `/test e2e-mixer-no_auth`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19532) | `/test istio-pilot-e2e-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@robertpanzer: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/release-test.sh | 5602476866a4666bab9c2380c7560d47b88ef29c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/release-test/6871) | `/test release-test`
prow/istio-unit-tests.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-unit-tests/24864) | `/test istio-unit-tests`
prow/e2e-simpleTests-cni.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-simpleTests-cni/3841) | `/test e2e-simpleTests-cni`
prow/e2e_pilotv2_auth_sds.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio_auth_sds_e2e/5857) | `/test istio_auth_sds_e2e`
prow/e2e-mixer-no_auth.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-mixer-no_auth/16502) | `/test e2e-mixer-no_auth`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19532) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-multicluster-e2e/11972) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@robertpanzer: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-unit-tests/24864) | `/test istio-unit-tests`
prow/e2e-simpleTests-cni.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-simpleTests-cni/3841) | `/test e2e-simpleTests-cni`
prow/e2e_pilotv2_auth_sds.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio_auth_sds_e2e/5857) | `/test istio_auth_sds_e2e`
prow/e2e-mixer-no_auth.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/e2e-mixer-no_auth/16502) | `/test e2e-mixer-no_auth`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-e2e-envoyv2-v1alpha3/19532) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/istio-pilot-multicluster-e2e/11972) | `/test istio-pilot-multicluster-e2e`
prow/release-test.sh | 3d0e484073d37811d822e0331f45534d23252fdf | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/13843/release-test/6901) | `/test release-test`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

robertpanzer commented:
Closing for now.
I can revisit this in case we would need support for machines with multiple IP addresses and services listening only on dedicated interfaces.
As long as we don't need this, the current behavior is fine.
